icicle/tests/test_hash_api.cpp

```diff
@@ -841,4 +1152,87 @@ TEST_F(HashApiTest, poseidon3_batch)
   ASSERT_EQ(0, memcmp(output_cpu.get(), output_cuda.get(), config.batch * sizeof(scalar_t)));
 }
 
-#endif // POSEIDON
\ No newline at end of file
+TEST_F(HashApiTest, poseidon_tree)
```

Comment:
why don't you use the 'test_merkle_tree' here?

Reply:
I can add it but here there is already comparison between cpu and gpu. I felt that adding the reference tree test redundant.